src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -87,7 +87,84 @@ public IWebHostBuilder CreateWebHostBuilder(IConsole console, string[] urls, str
                         services.ConfigureMetrics(context.Configuration);
                     }
                 })
-                .UseUrls(urls)
+                .ConfigureKestrel( (context, options) =>
+                {
+                    //Note our priorities for hosting urls don't match the default behavior.
+                    //Default Kestrel behavior priority
+                    //1) ConfigureKestrel settings
+                    //2) Command line arguments (--urls)
+                    //3) Environment variables (ASPNETCORE_URLS, then DOTNETCORE_URLS)
+
+                    //Our precedence
+                    //1) Environment variables (ASPNETCORE_URLS, DotnetMonitor_Metrics__Endpoints)
+                    //2) Command line arguments (these have defaults) --urls, --metricUrls
+                    //3) ConfigureKestrel is used for fine control of the server, but honors the first two configurations.
+
+                    string hostingUrl = context.Configuration.GetValue<string>(WebHostDefaults.ServerUrlsKey);
+                    if (!string.IsNullOrEmpty(hostingUrl))
+                    {
+                        urls = ConfigurationHelper.SplitValue(hostingUrl);
+                    }
+
+                    if (metrics)
```

Comment:
I think you need to lookup Metrics__Enabled from configuration rather than looking at the metrics parameter.

Since you would now be using two of the metrics configuration settings, might be easier to do (and less prone to error during refactors):
```
MetricsOptions metricsOptions = new MetricsOptions();
context.Configuration.Bind(MetricsOptions.ConfigurationKey, metricsOptions);
```
to get the values.

Reply:
This might require you to unconditionally call ConfigureMetricsEndpoint (with the minor adjustments I previously mentioned) so that the metrics and metricsUrls parameters are considered correctly.